### ISSUE TITLE
Endogenously calculate process emissions

### ISSUE THREAD

**Issue by robbieorvis** [Energy-Innovation/eps-us#189]
Now that the model tracks output of the main manufacturing industries, we should consider to endogenously calculating process emissions, which would free us from relying on updates to the Non-CO2 report that fail to account for updated GDP projections.  EPA actually takes this approach with estimating annual GHG emissions from industries: they use historical emissions rate per ton or dollar of output and then scale this based on newer data on output/production. We could easily do this for most process emissions.

For others that are less directly tied to output, we could develop some other methodologies:

1) For natural gas systems, we could use a leakage rate
2) For HFCs, we could build in a simplified consumption-emissions model like I've built for China (most accurate and best representation of policies like Kigali).
3) For waste, could scale by population or develop a landfilling model (EPA has several publicly available versions), etc...

We'd need to find a way to modify PERAC, but we could probably do this by embedding the scaling methodology we use within Vensim instead of doing all the work in Excel. This has some appeal since it would be much easier than the current approach in Excel and all the modifications could happen within Vensim.

**Comment by jrissman** [comment]
Partners we have worked with in the past (I'm thinking of WRI China here) have sometimes had their own process emissions inventories that they wanted to use.  This requires the ability to enter arbitrary process emissions via an input data file.

The EPA non-CO2 report is not so easily discarded, because it underlies PERAC as well as BPE.  Even partners that had their own process emissions projections never had their own marginal abatement cost curves.  As far as I'm aware, the EPA non-CO2 report is the unique, only source for this type of information on a country-by-country basis with global coverage.

This led to the development of a multiplier system that keeps BPE and PERAC synced up.  If you have your own data you want to use for BPE (i.e., because you think the EPA non-CO2 report's GDP projections are outdated), you can use your own data for BPE.  In BPE, you calculate multipliers (ratios between your data and the EPA data), and use those ratios in PERAC.  It is a refined system that has gone through many iterations and a great deal of work, and has been producing reasonable results for us for years.

Yes, Vensim could take a historical year of process emissions and scale up by output, presumably using data from the existing variable `BPCiObIC BAU Percent Change in Output by ISIC Code`.  But that doesn't simplify things because you'd still need to obtain the multipliers between your data and the EPA non-CO2 report's data, and then use them in the PERAC data files.  So you'd be doing the same calculations in Excel anyway, and this would introduce the risk that what you do in Excel doesn't exactly match what Vensim does internally.  Rather than reducing complexity or saving time, it just adds another potential point where bugs can come in.

You might think to eliminate the multipliers by also scaling PERAC by output (instead of by the multipliers generated in BPE). 
 Unfortunately, that approach is invalid because the non-CO2 report behind PERAC builds in changing technological availability over time, so scaling up the early year values by output would dramatically under-count available mitigation potential.

The existing BPE/PERAC system already is designed to support what you need.  All you need to do is to take a historical year of data for process emissions and scale it forward in Excel based on output projections.  Scaling historical data based on output is no great burden in Excel, and we do it in a number of other variables.  Then you take the ratios with the non-CO2 report's data to find the multipliers, and you past-as-values those multipliers into PERAC as normal (i.e. just like WRI did).  I do not, at present, see a way to improve on the simplicity or polish of the system we've already developed, because of the tight linkage between BPE and PERAC, and the need to continue to use the EPA non-CO2 report for PERAC.

**Comment by mkmahajan** [comment]
As a somewhat related note on potential edits to BPE and PERAC, I'm running into the need to tweak the current multiplier set-up to use annual multipliers rather than multipliers in 5 year increments. I think this is doable, it will just lead to some lengthy formulas in PERAC where we need to interpolate between years. Here's why I think this would be necessary for the US:

- We realized that since moving to AEO 2021 for the US, which accounts for COVID impacts on service demand, we had not adjusted BPE/PERAC to account for the drop in process emissions in 2020. We continued to simply take EPA's unadjusted non-CO2 data for the year 2020. I have now calculated multipliers for years 2020-2025 based on industrial output from AEO 2021, but our multipliers are only set up in 5 year increments. Using our current multiplier system would lead to a linear, 5-year recovery between 2020-2025 rather than what we actually see (a steep dip in 2020, followed by a pretty rapid recovery over the next 1-2 years). 

- Given the EPA's HFC announcement last week, we'd like to account for the AIM Act in our BAU. Rhodium has good data available for emissions under the AIM Act, and their trajectory has some sharp dips in years 2024, 2029, and 2034 that follow the Kigali phase-down schedule. But again, these sudden drops would be lost in our current multiplier system. This is another place where we would benefit from annual multipliers. Note that like we've done for a SNAP adjustment in the past, we will customize the abatement potential for the US after shifting our HFC baseline in BPE (but can keep this as a lever in the Country Selector that is only enabled for the US).

Jeff - I'm happy to take a stab at this and run it by you. We're trying to move pretty fast here on the federal modeling we're doing, so I'm trying to get a file set up for our us-analysis repo now. I may have to circle back after to get the structure set up to work fully with all regions.

**Comment by jrissman** [comment]
Certainly, it would be possible to switch to annual multipliers.  The current approach uses multipliers at 5-year increments because the data in PERAC are given in 5-year increments, and we did not anticipate a need to model rapid year-over-year changes near to the present day.

It should be relatively straightforward to switch to annual multipliers in the BPE and PERAC Excel files.  In fact, BPE already uses annual increments in the formulas in its blue tabs (it converts the 5-year multipliers to annual multipliers), so it should be particularly easy to switch this file.  Just calculate the multipliers directly in the annual section rather than in the 5-year section (and then you can delete the 5-year section).  No updates to the blue tabs should be needed, and I don't think it should interact with or harm the file's customizability to different countries or regions.

PERAC is slightly more bother because, as you mention, you would need to update the interpolation formulas, so they interpolate between the values prior to applying any multipliers, and then multipliers are applied on an annual basis after the interpolation is done.  Still, it doesn't sound too bad.  Again, I don't see it interacting with the file's customizability to different regions.

One small thing to keep in mind is to keep the format of the multipliers table identical between BPE and PERAC, since the normal procedure is to calculate the multipliers in BPE, then use paste-as-values to transfer those multipliers to PERAC.

Thanks for offering to tackle this data-only edit.  Would it make sense to do the update officially in the "develop" branch of the us-eps repo and then copy that file over to your special analysis branch for further customization, if needed?  (Robbie has indicated we are ultimately going to publish this work and will want to be able to show the scenarios in the public webtool, so this feature - annual multipliers - will need to be part of an official US-EPS release anyway.)

**Comment by jrissman** [comment]
Update:

BPE and PERAC have been switched to use annual multipliers rather than 5-year multipliers.

We think it might be worthwhile to calculate process emissions endogenously, as long as there is an override BPE file for partners that already have a process emissions inventory, which they demand be used.

To keep in sync with PERAC, and to eliminate the difficult task of maintaining PERAC as an Excel file, we are interested in potentially doing more of the PERAC adjustment calculations in Vensim and fewer in Excel, but we likely still need to do some PERAC calculations in Excel.  For instance, the EPA source doesn't apportion anything into price bins, whereas we both apportion into price bins and allocate technological measures to specific policy levers.  Those tasks might need to continue to be done in Excel.

**Comment by robbieorvis** [comment]
I'm changing this to high priority. It will be necessary once we start to allow users to switch between different production technologies (like in iron and steel) to correctly calculate process emissions reductions.

**Comment by robbieorvis** [comment]
I used Claude to suggests some approaches here. Findings below:

# Endogenize Process Emissions Baseline (BPE) and MACC (PERAC) in EPS

## Background

EPS 4.0.5 treats industrial process emissions almost entirely as exogenous inputs:

- **`BPE BAU Process Emissions[Industry Category, Pollutant]`** (EPS.mdl:15838–15921) — one CSV per pollutant under `InputData/indst/BPE/`, absolute g pollutant by ISIC × year (2020–2050).
- **`PERAC Mass CO2e Avoidable by Marginal Cost[Proc Emis Abatement Category, Marginal Abatement Cost]`** (EPS.mdl:41485–41632) — a pre-bucketed MACC on a fixed 72-bin price grid (`PriceBin1..PriceBin72`, $50/ton wide, –$1,125 to +$2,475; bin→$ in `PERAC-MCD.csv`), fixed quantity per year.
- **`PEAAC Mass CO2e Additions and Costs`** (EPS.mdl:41165–41307) — $ cost data on the same grid.

Current scaling logic (`BAU Process Emissions after GDP and Fuel Production Adjustment`, EPS.mdl:10036–10049):

- **Non-energy industries**: `BPE × (1 + Perc Change in Sectoral Energy Use due to Exogenous GDP Adjustment[industry sector])` — only responds to exogenous GDP.
- **Energy industries** (coal 05, O&G 06, refined petroleum 19, pipelines/gas processing 352T353): `BPE × (1 + Last Year BAU Percent Change in Fuel Production from Calculated Fuel Demand)` — already tied to endogenous fuel production.

Deployment of abatement uses two routes: carbon-tax break-even (`Process Emissions Abated due to Carbon Tax by Marginal Cost`, EPS.mdl:43862+) and direct policy intensity via `Selected Policy Implementation Schedule`.

## Problems this creates

1. Non-energy BPE is unresponsive to endogenous physical output — cement, steel, chemicals emissions drift if the IO module says those industries grow/shrink faster than the baseline assumed.
2. PERAC is a **fixed-quantity** curve: if baseline cement emissions double endogenously, available abatement does not. EPA's non-CO2 MACC methodology in contrast has abatement scale proportionally with baseline (`applicability × market share × reduction efficiency × baseline`).
3. Process-CO2 coverage is sparse. The 17 abatement categories map to 8 ISIC codes; cement is the only process-CO2 lever. **Iron & steel, ammonia, petrochemicals, lime, aluminum, limestone/dolomite use have no structural lever.**
4. No representation of **stoichiometric floor vs. route-swappable** emissions (calcination CO2 is chemically fixed; SMR-H2 vs electrolytic H2 is not).
5. F-gas inventory only has end-of-pipe measures — no HFC vintaging / Kigali phase-down structure tied to end-use substitution.
6. Updating the MACC requires manually re-bucketing external sources into 72 price bins × 31 years for every abatement category, annually.

## What EPA / IPCC do (relevant methodology takeaways)

**EPA Global Non-CO2 Projections (EPA-430-R-19-012, 2019)** — 195 countries, 1990–2050.

- Baseline = IPCC Tier 1 `activity × fixed EF`, source-by-source, country-by-country. Activity drivers are physical (coal production, head of livestock, waste-in-place FOD, HFC consumption vintaged across ~65 end uses).
- MACCs are option-level: each option has `applicability% × market share × reduction efficiency × baseline`, plus capital + O&M + revenue + lifetime at 5% discount rate. Break-even $/tCO2e sets the x-axis position; abatement **auto-scales with baseline**.
- 2019 update adds learning curves (15% rate) and +1%/yr reduction-efficiency improvement.

**Process CO2 (EPA IPPU chapter + IPCC 2006 Vol 3 + 2019 Refinement).** Dominant US IPPU CO2 sources, descending: iron & steel ≈ cement » petrochemicals > lime > ammonia > other limestone/dolomite > aluminum > urea ≈ soda ash > ferroalloys > glass/TiO2/etc. Key idea: split each source into a **chemical floor** (CaCO3 → CaO; urea-N hydrolysis; addressable only via CCS) and **route-swappable** emissions (BF-BOF → H2-DRI-EAF; SMR NH3 → electrolytic; naphtha crackers → e-cracking; prebake Al → inert anode). Route-swap levers are absent from EPS today.

## International data reproducibility (hard requirement)

Since EPS runs globally, every new structural choice needs a data source we can use outside the US.

| Component | Global source | Status |
|---|---|---|
| Non-CO2 MACCs | EPA 2019 non-CO2 projection (195 countries) | ✅ authoritative global dataset |
| Non-CO2 baselines | Same EPA dataset + EDGAR for anchoring | ✅ |
| Cement clinker production | USGS MCS + Andrew 2019 global clinker series (ESSD, open) | ✅ |
| Steel by route | WorldSteel Statistical Yearbook | ✅ |
| Aluminum | International Aluminium Institute stats | ✅ |
| Ammonia | FAOSTAT + USGS; IFA is paywalled | ✅ (basic) |
| Lime | USGS (limited ex-US) — fallback: scale to cement | ⚠️ proxy |
| Petrochemicals | IEA ETP is standard; no clean open global series | ⚠️ **gap** — fallback: scale to industrial GDP + EDGAR 2.B intensity |
| HFCs consumption | UNEP Ozone Secretariat Article 7 (Kigali reporting) | ✅ by species, no end-use split |
| HFCs end-use shares | Velders et al. (2015, 2022 — *PNAS*, *ACP*): 6 end uses globally | ✅ coarser than US EPA Vintaging Model's 65 |
| F-gas inventory | EDGAR v8.0 F-gases by country × IPCC 2-digit | ✅ |
| Historical anchor, full time series | **EDGAR v8.0** (all countries, all categories, CO2+CH4+N2O+F-gas, 1970–2023, open) | ✅ **primary** |
| Historical anchor, facility-level | **Climate TRACE** (2015–2024, strong on cement/steel/O&G, weaker on ag/waste/F-gas) | ✅ cross-validation |
| Alt historical reconciliation | PRIMAP-hist (EDGAR + UNFCCC CRF + CDIAC) | ✅ optional |

**Key constraint:** Phase 3 (HFC vintaging) must be scoped to the **Velders 6-end-use structure**, not the EPA Vintaging Model's ~65 end uses, to be buildable for any country from UNEP + EDGAR + GDP. That's the only real scope adjustment from a US-only version of the plan.

## Proposed structural model

Core equation per source:

```
ProcessEmissions[src, gas] = Σ_route ActivityOutput[src] × RouteShare[src, route]
                           × StoichiometricEF[src, route, gas]
                           × (1 − SubstitutionShare[src])    // clinker-to-cement, scrap share, feedstock swap
                           × (1 − CCS_CaptureRate[src])
```

Two endogenous activity multipliers already in EPS to reuse:

- `Output by ISIC Code by Year[ISIC Code]` (io-model) — use for non-energy industries.
- `Last Year BAU Percent Change in Fuel Production from Calculated Fuel Demand` — already wired for energy industries; extend to `Fuel Production Mapped to ISIC Codes` (EPS.mdl:25949).

Replace fixed-quantity PERAC bins with **parameterized options** — one record per (source, lever) with: applicability fraction, max reduction, capital $/unit, fixed O&M, variable O&M, revenue stream, lifetime, learning rate, start year, ramp rate. Quantity available is computed each step from current baseline × applicability × max reduction; break-even $ is computed from current fuel/electricity prices and post-learning capex. Abatement auto-scales with baseline, matching EPA methodology.

## Phased plan

### Phase 1 — Endogenize BPE driver (no new data, ~1 week)

- Change the non-energy branch at EPS.mdl:10036–10049 from the industry-GDP scalar to `Output by ISIC Code by Year / initial-year Output`. Mirror on the non-BAU path (`Process Emissions after CCS`).
- Exception list for sources that shouldn't scale with ISIC $-output:
  - Livestock CH4/N2O → herd proxy (FAOSTAT globally)
  - Rice CH4 → crop area (FAOSTAT)
  - Waste CH4 → IPCC FOD driven by population × per-capita MSW
  - HFCs → deferred to Phase 3
- **Known limitation**: `Output by ISIC Code by Year` is a value measure, not physical. Document this; address later if needed by deflating with sector-specific prices.

### Phase 2 — Split EF into floor + variable; add route share for process CO2 (~3–4 weeks)

Add a `Process Route` subscript *or* per-source boolean shares (subscript is cleaner but costs cells on cross-products; boolean duplicates equations). Priority order:

1. **Cement** — (clinker × 0.51 t/t) + clinker-to-cement ratio + CCS rate. Retire existing PERAC `cement measures` category; replace with three levers (clinker substitution %, SCM cap, CCS share).
2. **Iron & steel** — route shares BF-BOF / scrap-EAF / NG-DRI-EAF / H2-DRI-EAF, EFs ~1.8 / 0.08 / 1.0 / 0.05 t CO2/t steel. Policy lever = target H2-DRI share. **New** lever — not in PERAC today.
3. **Ammonia** — SMR vs electrolytic, EF 1.66 vs ~0 t CO2/t NH3. Lever = green NH3 share. Tie to existing hydrogen pathway module.
4. **Petrochemicals** — ethylene/methanol/carbon black feedstock route + CCS on cracker vents.
5. **Lime, limestone use, aluminum, ferroalloys, glass** — add CCS rate as the only lever (calcination chemically fixed).

For non-CO2 sources already in PERAC: keep the abatement-category structure, but change `PERAC Mass CO2e Avoidable` from a fixed annual time series to `applicability × max_reduction × baseline_this_year`.

### Phase 3 — HFC vintaging with Velders-style structure (~2 weeks)

Replace BPE F-gas for chemicals 20 + other metals 242 with a minimal vintaging stock:

- 6 end uses (MAC, stationary AC, refrigeration, foams, aerosols, solvents + fire) à la Velders et al.
- 3–4 substitute generations per end use.
- Kigali phase-down schedule as the policy lever.
- Driven by UNEP Article 7 consumption × GDP-per-capita regressors (Velders approach) for non-US; US can optionally use EPA Vintaging Model output as a cross-check.

F-gas is ~49% of US IPPU CO2e, so impact/effort ratio is high.

### Phase 4 — Replace pre-bucketed PERAC CSVs with computed MACC (~2 weeks)

- Ship one `ProcEmisLevers.csv` per source: `source, lever_id, applicability%, max_reduction%, CC_$/unit, fixed_OM, variable_OM, revenue_type, lifetime_yr, learning_rate, start_year, ramp_years`.
- Compute break-even $/tCO2e each step from current fuel/electricity prices and post-learning capex.
- Drop the 72-bin structure; the carbon-tax gate (EPS.mdl:43862) becomes `IF tax > break_even_$`.
- Regenerate option parameters from EPA's published global non-CO2 option tables + Rhodium/CATF/DOE Liftoff for process CO2.
- **Important**: recompute MACC bins per year, not per Optimization Pass, to avoid convergence loops.

### Phase 5 — Historical anchoring + validation

- Anchor 2015–2023 emissions to **EDGAR v8.0** globally (EPA GHG Inventory for US cross-check). Use Climate TRACE for facility-level validation on cement, steel, O&G methane.
- Back-test: feed AEO 2023 US industrial output into the endogenous structure, reproduce EPA inventory within ~5% sector-by-sector.
- Cross-check F-gas projection against Velders published scenarios (non-US) and EPA Vintaging Model (US).

## Open questions before starting

1. **Subscript design**: add `Process Route` as a new subscript (clean; cross-product cost) vs. per-source boolean shares (avoid cross-product; duplicated equations). Preference?
2. **Activity driver resolution**: accept dollar-valued `Output by ISIC Code by Year` for Phase 1, or build physical-output variables for priority sources now?
3. **HFC module location**: new `InputData/hfcs/` folder, or extension of `InputData/indst/`? It crosses buildings (AC/refrigeration) and industry — may want its own home.
4. **Backward compatibility**: keep BPE CSVs as a per-source fallback toggle so sources can flip to endogenous one at a time?
5. **Carbon-tax path dependency**: confirm annual (not per-pass) MACC recomputation to keep the Optimization Pass loop convergent.

## References

- EPA-430-R-19-012: Global Non-CO2 GHG Emission Projections & MAC Analysis: Methodology Documentation (2019) — https://www.epa.gov/sites/default/files/2019-09/documents/nonco2_methodology_report.pdf
- EPA Inventory of U.S. GHG Emissions and Sinks, IPPU Chapter (latest edition) — https://www.epa.gov/ghgemissions/inventory-us-greenhouse-gas-emissions-and-sinks
- IPCC 2006 Guidelines Vol 3 (IPPU); 2019 Refinement
- Andrew (2019) — Global CO2 emissions from cement production, *Earth System Science Data*
- Velders et al. (2015, 2022) — global HFC scenarios, *PNAS* / *Atmos. Chem. Phys.*
- EDGAR v8.0 — JRC Joint Research Centre
- Climate TRACE — climatetrace.org
- PRIMAP-hist — https://www.pik-potsdam.de/paris-reality-check/primap-hist/
- IPCC AR6 WGIII Chapter 11 (Industry)
- DOE Industrial Decarbonization Liftoff
